### PR TITLE
Deleted duplicate action results in WebapiCompatShim

### PR DESCRIPTION
**Pull request by ajaybhargavb**
@rynowak

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.WebApiCompatShim/ApiController.cs [hunk 1]
```diff
@@ -93,16 +91,6 @@ public IPrincipal User
...
 
         /// <summary>
-        /// Creates a <see cref="BadRequestResult"/> (400 Bad Request).
-        /// </summary>
-        /// <returns>A <see cref="BadRequestResult"/>.</returns>
-        [NonAction]
-        public virtual BadRequestResult BadRequest()
-        {
-            return new BadRequestResult();
-        }
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.WebApiCompatShim/ApiController.cs]: These methods should still be here and return the new result types. ApiController does not extend Controller

**Comment by rynowak** [comment]
:watch: Deleted too much! Just the action result types should go. All of the functionality needs to stay

**Comment by ajaybhargavb** [comment]
Updated.

File: test/Microsoft.AspNet.Mvc.WebApiCompatShimTest/ActionResults/CreatedAtRouteResultTest.cs [hunk 2]
```diff
@@ -13,10 +13,10 @@
 
 namespace System.Web.Http
 {
-    public class CreatedAtRouteNegotiatedContentResultTest
+    public class CreatedAtRouteResultTest
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.WebApiCompatShimTest/ActionResults/CreatedAtRouteResultTest.cs]: Do we need these tests? Are they duplicates of tests we already have for MVC?

File: test/Microsoft.AspNet.Mvc.WebApiCompatShimTest/ActionResults/CreatedResultTest.cs [hunk 3]
```diff
@@ -13,10 +13,10 @@
 
 namespace System.Web.Http
 {
-    public class CreatedNegotiatedContentResultTest
+    public class CreatedResultTest
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.WebApiCompatShimTest/ActionResults/CreatedResultTest.cs]: Do we need these tests? Are they duplicates of tests we already have for MVC?

File: test/Microsoft.AspNet.Mvc.WebApiCompatShimTest/ApiControllerTest.cs [hunk 4]
```diff
@@ -161,17 +162,21 @@ public void ApiController_CreatedAtRoute_Dictionary()
             var controller = new ConcreteApiController();
 
             var product = new Product();
-            var values = new RouteValueDictionary(new { controller = "Products" });
+            var expected = new Dictionary<string, object>
+                {
+                    { "controller", "Products" },
+                };
+            var values = new RouteValueDictionary(expected);
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.WebApiCompatShimTest/ApiControllerTest.cs]: why this change?

**Comment by rynowak** [comment]
:watch:

File: test/Microsoft.AspNet.Mvc.WebApiCompatShimTest/ApiControllerTest.cs (hunk 4 above)
Comment (ajaybhargavb) [inline test/Microsoft.AspNet.Mvc.WebApiCompatShimTest/ApiControllerTest.cs]: Because directly asserting with `values` fails.
> Assert.Equal() Failure
Expected: RouteValueDictionary [[controller, Products]]
Actual:   Dictionary`<String, Object>` [[controller, Products]]

**Comment by ajaybhargavb** [comment]
Updated.

File: test/Microsoft.AspNet.Mvc.WebApiCompatShimTest/ApiControllerTest.cs (hunk 4 above)
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.WebApiCompatShimTest/ApiControllerTest.cs]: use `Assert.Equal<KeyValuePair<string, object>>(...)` 

The generic type parameter forces xunit to treat these as IEnumerables.

**Comment by rynowak** [comment]
:shipit:
